aurelia-slickgrid/src/aurelia-slickgrid/services/filter.service.ts

```diff
@@ -378,12 +383,21 @@ export class FilterService {
   }
 
   /**
-   * A simple function that is attached to the subscriber and emit a change when the sort is called.
+   * A simple function that will be called to emit a change when a filter changes.
    * Other services, like Pagination, can then subscribe to it.
-   * @param {string} sender
+   * @param sender
    */
-  emitFilterChangedBy(sender: string) {
-    this._subscriber.subscribe(() => this.onFilterChanged.publish('filterService:changed', `onFilterChanged by ${sender}`));
+  emitFilterChanged(sender: 'local' | 'remote') {
```

Comment:
if the `sender` can only be `local` or `remote` then you may be able to simplify the this logic. Or are you playing it safe? The simplification can be:

```typescript
  let currentFilters: CurrentFilter[] = [];

  if (sender === 'remote') {
      const backendService = this._gridOptions.backendServiceApi.service;
      if (backendService && backendService.getCurrentFilters) {
        currentFilters = backendService.getCurrentFilters() as CurrentFilter[];
      }
    } else {
      currentFilters = this.getCurrentLocalFilters();
    }

  this.ea.publish('filterService:filterChanged', currentFilters);
```

Reply:
I'm playing it safe and it's still better to see what is acceptable right from the argument